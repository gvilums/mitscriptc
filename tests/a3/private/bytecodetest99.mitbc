function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [" foobar ", None],
			parameter_count = 2,
			local_vars = [a, b],
			local_ref_vars = [],
			free_vars = [],
			names = [print],
			instructions = 
			[
				load_global	0
				load_local	0
				load_const	0
				add
				load_local	1
				add
				call	1
				pop
				load_const	1
				return
			]
		},
		function
		{
			functions = [],
			constants = [1, None],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [z, f],
			instructions = 
			[
				load_global	0
				load_const	0
				add
				store_global	0
				load_global	1
				load_global	0
				load_global	0
				call	2
				pop
				load_const	1
				return
			]
		}
	],
	constants = [1, "foobar", "foo", "bar", "nope", 5, false, 0],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [x, b, r, f, z, f2, y, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	7
		load_func	1
		alloc_closure	0
		store_global	8
		load_func	2
		alloc_closure	0
		store_global	9
		load_const	0
		neg
		store_global	0
		load_const	1
		store_global	1
		alloc_record
		dup
		dup
		load_const	2
		field_store	6
		load_const	3
		field_store	0
		store_global	2
		load_global	2
		load_global	2
		eq
		not
		if	6
		load_global	7
		load_global	2
		call	1
		pop
		goto	5
		load_global	7
		load_const	4
		call	1
		pop
		load_func	3
		alloc_closure	0
		store_global	3
		load_global	0
		load_const	5
		swap
		gt
		not
		if	13
		load_global	3
		load_global	0
		load_global	0
		load_const	0
		sub
		call	2
		pop
		load_global	0
		load_const	0
		add
		store_global	0
		goto	-17
		load_const	6
		not
		if	6
		load_global	0
		load_const	7
		div
		store_global	0
		goto	-7
		load_const	0
		store_global	4
		load_func	4
		alloc_closure	0
		store_global	5
		load_global	7
		load_global	5
		call	0
		call	1
		pop
		load_global	7
		load_global	4
		call	1
		pop
	]
}