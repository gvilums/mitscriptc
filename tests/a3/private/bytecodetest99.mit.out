{x:bar y:foo }
-1 foobar -2
0 foobar -1
1 foobar 0
2 foobar 1
3 foobar 2
4 foobar 3
2 foobar 2
None
2
